src/components/VirtualList/useItemsPerPage.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+
+export function calculateItemsPerPage(
+  containerHeight: number,
+  itemHeight: number
+) {
+  // Throwing on an extra `1` just to play it safe
+  return Math.ceil(containerHeight / itemHeight) + 1;
```

Comment:
react-virtualized has a row overscan prop that allows extra rows off screen, and a prop to start fetching next page when X rows close to the bottom. This allowed the buffer to be configured based on usage to help make the loading more seamless.

Reply:
Derp, it turns out `react-infinite-scroll-component` has this as well, and it defaults to 80% of the height of the container. So we probably don't need the extra element.